### PR TITLE
UTs: Refactor to global usings

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: Let's add also
* System
* System.Collections.Generic
* System.Collections.Immutable
* System.Linq
* Microsoft.CodeAnalysis
* SonarAnalyzer.Extensions (there are not too many yet, but should be increasing while we move stuff from helpers)
* SonarAnalyzer.Helpers
* Moq

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: And `Microsoft.CodeAnalysis.Diagnostics`

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: These are just from top of my head. Did you try to write a small scraper that would measure what are the most often used ones? (Except C# and VB specific)

File: analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+global using FluentAssertions;
+global using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:22]: https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#using

Can it be done in `csproj` file like in VB? Having it hidden/scattered in a random file somewhere in the project is a quite poor language design decision
```
<ItemGroup>
  <Using Include="System.Text.Json" />
  <Using Include="System.Text.Json.Serialization" />
</ItemGroup>
```

I didn't try if it works.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Just moving the card, Costin is off today

File: analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: Oh, and the `SonarAnalyzer.UnitTest.MetadataReferences`

Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: > These are just from top of my head. Did you try to write a small scraper that would measure what are the most often used ones? (Except C# and VB specific)

using System.Security; is used 42 times.
using System.Web; is used 46 times.
using System.Xml.XPath; is used 46 times.
using Microsoft.AspNetCore.Http; is used 46 times.
using System.ComponentModel; is used 48 times.
using Xunit; is used 56 times.
using Microsoft.AspNetCore.Mvc; is used 61 times.
using System.Net; is used 67 times.
using NUnit.Framework; is used 68 times.
using System.Collections.Immutable; is used 73 times.
using System.Globalization; is used 74 times.
using System.Collections.ObjectModel; is used 79 times.
using SonarAnalyzer.Common; is used 89 times.
using System.Runtime.CompilerServices; is used 90 times.
using SonarAnalyzer.Helpers; is used 95 times.
using System.Reflection; is used 96 times.
using Microsoft.VisualStudio.TestTools.UnitTesting; is used 106 times.
using SonarAnalyzer.UnitTest.MetadataReferences; is used 121 times.
using System.Threading; is used 123 times.
using System.Security.Cryptography; is used 131 times.
using CS = SonarAnalyzer.Rules.CSharp; is used 132 times.
using VB = SonarAnalyzer.Rules.VisualBasic; is used 133 times.
using System.Diagnostics; is used 139 times.
using System.Xml; is used 145 times.
using Microsoft.CodeAnalysis; is used 149 times.
using System.Runtime.Serialization; is used 154 times.
using System.Runtime.InteropServices; is used 173 times.
using System.Collections; is used 243 times.
using SonarAnalyzer.Rules.CSharp; is used 268 times.
using System.Text; is used 281 times.
using System.IO; is used 399 times.
using System.Threading.Tasks; is used 493 times.
using System.Linq; is used 727 times.
using System.Collections.Generic; is used 1970 times.
using System; is used 4495 times.

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: This doesn't seem right, it probably includes TestCases directory too. `System.Runtime.Serialization` was found only 32x by VS for example.

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs:20]: `System.Threading.Tasks` is 493x above and only 4x in the project. If you have some utility for it, you can do something like `code.Substring(0, code.IndexOf("namespace"))`

Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/GlobalUsings.cs]: > Let's add also
> 
> * System
> * System.Collections.Generic
> * System.Collections.Immutable
> * System.Linq
> * Microsoft.CodeAnalysis
> * SonarAnalyzer.Extensions (there are not too many yet, but should be increasing while we move stuff from helpers)
> * SonarAnalyzer.Helpers
> * Moq

Added them except SonarAnalyzer.Extensions and Moq, as those two would create ambiguity between some other classes and methods from other namespaces with the same names.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/VbcHelperTest.cs [hunk 3]
```diff
@@ -49,9 +45,7 @@ public class VbcHelperTest
         [DataRow("   eRrOr fOo:", true)]
         [DataRow("   error in foo:", false)]
         [DataRow("   error in foo :", false)]
-        public void IsTextMatchingVbcErrorPattern_ReturnsExpected(string text, bool expectedResult)
-        {
+        public void IsTextMatchingVbcErrorPattern_ReturnsExpected(string text, bool expectedResult) =>
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/VbcHelperTest.cs:48]: _No action needed:_ This is unrelated, it's better to limit scope of such big PRs
